### PR TITLE
Update EventError with Is and Unwrap functions

### PR DESCRIPTION
**Pull request by hamersaw**
Signed-off-by: Daniel Rammer <daniel@union.ai>

# TL;DR
The golang errors package provides the functions Is (to check if an instance of the error type is within the stack), As (to convert the error to the error type if it is within the stack), and Unwrap (to facilitate the aforementioned functionality). This updates the EventError type to enable use within the API.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
See TL;DR

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: clients/go/events/errors/errors.go [hunk 1]
```diff
@@ -31,6 +32,18 @@ func (r EventError) Error() string {
 	return fmt.Sprintf("%s: %s, caused by [%s]", r.Code, r.Message, r.Cause.Error())
 }
 
+func (r *EventError) Is(target error) bool {
+	t, ok := target.(*EventError)
+	if !ok {
+		return false
+	}
+	return r.Code == t.Code && (r.Cause == t.Cause || t.Cause == nil) && (r.Message == t.Message || t.Message == "")
```
Comment (EngHabu) [inline clients/go/events/errors/errors.go:40]: Should that just check for the Code? I feel the message and cause may be different but the errors will still be _the same_... it's not an "Equals" check.. or at least in my opinion it doesn't have to be...

Comment (hamersaw) [reply clients/go/events/errors/errors.go:40]: Good point. I saw it implemented this way elsewhere and thought it might add more extensibility. The idea is that when checking for an error if you don't provide the Cause or Message than it performs the check as you suggested - equality over Code and then Cause == nil / Message == "". This scenario is applied in the updates to IsNotFound, IsAlreadyExist, etc below. For example:

`errors.Is(err, &EventError{Code: ExecutionNotFound}`

Only checks if the Code is equal to any error in the stack. However, if a user wanted to include more fine-grained checks they could include a Cause or Message as well. I don't have strong feelings either way, if you prefer to perform only a Code check I can make the update.

Comment (EngHabu) [reply clients/go/events/errors/errors.go:40]: SG 👍🏽

File: clients/go/events/errors/errors.go [hunk 2]
```diff
@@ -31,6 +32,18 @@ func (r EventError) Error() string {
 	return fmt.Sprintf("%s: %s, caused by [%s]", r.Code, r.Message, r.Cause.Error())
 }
 
+func (r *EventError) Is(target error) bool {
+	t, ok := target.(*EventError)
+	if !ok {
+		return false
+	}
```
Comment (EngHabu) [inline clients/go/events/errors/errors.go:39]: The one additional check I would add is:
```suggestion
	}
	
	if r == nil && t == nil {
	    return true
	}
	
	if r == nil || t == nil {
	    return false
	}
```

Example for why: https://play.golang.org/p/P85Pext7xw5

Comment (hamersaw) [reply clients/go/events/errors/errors.go:39]: Very interesting, it doesn't seem like the check should be necessary, but the example certainly clarifies it.

**Review by EngHabu — APPROVED** [review]

**Review by wild-endeavor — APPROVED** [review]
is this backwards compatible with everything?

**Comment by hamersaw** [comment]
> is this backwards compatible with everything?

@wild-endeavor Yes, I just added a few functions to support traversal through the entire error stack to check for EventError types.
